common/persistence/nosql/nosqlPersistenceTest.go

```diff
@@ -32,27 +32,35 @@ import (
 
 // testCluster allows executing cassandra operations in testing.
 type testCluster struct {
-	keyspace string
-	cfg      config.NoSQL
+	keyspace  string
+	schemaDir string
+	cfg       config.NoSQL
 }
 
 var _ testcluster.PersistenceTestCluster = (*testCluster)(nil)
 
 // NewTestCluster returns a new cassandra test cluster
-func NewTestCluster(pluginName, keyspace, username, password, host string, port int, protoVersion int) testcluster.PersistenceTestCluster {
-	var tc testCluster
-	tc.keyspace = keyspace
-	tc.cfg = config.NoSQL{
-		PluginName:   pluginName,
-		User:         username,
-		Password:     password,
-		Hosts:        host,
-		Port:         port,
-		MaxConns:     2,
-		Keyspace:     keyspace,
-		ProtoVersion: protoVersion,
+// if schemaDir is empty, it will be auto-resolved based on os.Getwd()
+// otherwise the specified value will be used (used by internal tests)
+func NewTestCluster(
+	pluginName, keyspace, username, password, host string,
+	port, protoVersion int,
+	schemaDir string,
```

Comment:
How about schemaBaseDir? it will sound more "optional" and more accurate based on the actual meaning.